src/components/Header/Header.tsx

```diff
@@ -0,0 +1,134 @@
+import {
+  useCallback,
+  useEffect,
+  useState,
+} from 'react';
+import {
+  useLocation,
+  useParams,
+} from 'react-router-dom';
+import classNames from 'classnames';
+import { Logo } from '../Logo';
+import { MenuButton } from '../MenuButton';
+import { PageNav } from '../PageNav';
+import { TopActionButton } from '../TopActionButton';
+import { SearchBar } from '../SearchBar';
+import { useWindowSize } from '../../utils/useWindowSize';
+import productCategory from '../../api/productCategory.json';
+import './Header.scss';
+
+export const Header: React.FC = () => {
+  const location = useLocation();
+  const { productId = '' } = useParams();
+
+  const size = useWindowSize();
```

Comment:
```suggestion
  const { width } = useWindowSize();
```
not obligatory, but you can use destructuring if you don't use entire `size` object

Reply:
No problem 😉 Done! ✅